src/org/jgroups/perf/AsyncCounterBenchmark.java

```diff
@@ -81,15 +87,26 @@ private void updateTime(long timeNanos) {
         }
     }
 
-    private CompletionStage<Void> updateCounter(AsyncCounter counter, long start) {
-        if (stop.get()) {
-            // we don't check the return value
-            return CompletableFutures.completedNull();
-        }
-        return counter.addAndGet(deltaSupplier.getAsLong()).thenCompose(__ -> {
-            final long currentTime = System.nanoTime();
-            updateTime(currentTime - start);
-            return updateCounter(counter, currentTime);
+    private void updateCounter(AsyncCounter counter) {
+        requests.increment();
+        long start = System.nanoTime();
+        counter.addAndGet(deltaSupplier.getAsLong()).thenRun(() -> {
+            long time = System.nanoTime();
+            updateTime(time - start);
+            queue.offer(time);
```

Comment:
TBH capturing latency shouldn't be a big deal...I suggest switching to using a non thread-safe HdrHistogram (in the other PR) so, given that I/O is happening on a single thread, there is no need for atomics or any synchronization mechanism